src/noit_metric_director.c

```diff
@@ -1114,7 +1115,13 @@ static mtev_boolean
 check_duplicate(char *payload, size_t payload_len) {
   if (dedupe) {
     unsigned char *digest = malloc(MD5_DIGEST_LENGTH);
-    MD5((unsigned char*)payload, payload_len, digest);
+    const EVP_MD *md = EVP_get_digestbyname("MD5");
```

Comment:
should make our own md5 function in a single file that gets called from these sites instead of propagating ``OpenSSL`` code throughout

Reply:
There is both an md5 and a sha1 use case though, so we'd need two, and that's the same number of times we're currently using it, so I don't see what that gets us.